api/Silk.Api/Services/SpotifyService.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Security.Authentication;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Silk.Api.ApiResponses.Spotify;
+
+namespace Silk.Api.Services
+{
+    public sealed class SpotifyService
+    {
+        private readonly string _clientId;
+        private readonly string _clientSecret;
+        private string _accessToken;
+        private string _refreshToken;
+        private DateTime? _tokenExpiry;
+        
+        private readonly HttpClient _spotifyHttpClient = new();
+        
+        private const string TokenisationUrl = "https://accounts.spotify.com/api/token";
+        private const string TrackLookupUrl = "https://api.spotify.com/v1/search";
+        private const string ArtistLookupUrl = "https://api.spotify.com/v1/artists";
+        
+        public SpotifyService(string clientId, string clientSecret)
+        {
+            _clientId = clientId;
+            _clientSecret = clientSecret;
+            
+            CheckAndRefreshTokens().GetAwaiter().GetResult();
+        }
+        
+        public async Task CheckAndRefreshTokens()
```

Comment:
Nitpick: Should be postfixed with `Async`

Reply:
Next commit